.infrastructure/security/rbac.yml

```diff
@@ -0,0 +1,29 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: secrets-reader
+  namespace: todoapp
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: secrets-reader
+  namespace: todoapp
+rules:
+  - apiGroups: [""]
+    resources: ["pods"]
```

Comment:
The current role only grants access to 'pods'. If the intention is to allow the service account to read secrets, you should include 'secrets' in the resources list. For example: resources: ["pods", "secrets"].